File: 2406-divide-intervals-into-minimum-number-of-groups/2406-divide-intervals-into-minimum-number-of-groups.cpp
class Solution {
public:
    int minGroups(vector<vector<int>>& ar) {
        vector<int> dp((int)1e6+2);
        for(int i=0;i<ar.size();i++){
            dp[ar[i][0]]+=1;
            dp[ar[i][1]+1]-=1;
        }
        int mx=0;
        for(int i=1;i<dp.size()-1;i++){
            dp[i]+=dp[i-1];
            mx=max(mx,dp[i]);
        }
        return mx;
    }
};

File: 2406-divide-intervals-into-minimum-number-of-groups/2406-divide-intervals-into-minimum-number-of-groups.java
class Solution {
    public int minGroups(int[][] ar) {
        int dp[] = new int[(int)1e6+2];
        for(int i=0;i<ar.length;i++){
            dp[ar[i][0]]+=1;
            dp[ar[i][1]+1]-=1;
        }
        int max=0;
        for(int i=1;i<dp.length-1;i++){
            dp[i]+=dp[i-1];
            max=Integer.max(max,dp[i]);
        }
        return max;
    }
}